lib/src/components/NxTextLink/__tests__/NxTextLink.test.tsx

```diff
@@ -5,114 +5,106 @@
-    expect(getShallow({ href: '#/pages/Text%20Link' })).toHaveProp('href', '#/pages/Text%20Link');
-    expect(getShallow({ href: '#/pages/Text%20Link', disabled: true })).toHaveProp('href', undefined);
+    expect(renderEl({ href: '#/pages/Text%20Link' })?.getAttribute('href')).toContain('#/pages/Text%20Link');
+    expect(renderEl({ href: '#/pages/Text%20Link', disabled: true })?.getAttribute('href')).toBe(null);
   });
 
   it('sets the target to _blank if newTab is true, unless a different target is specified', function() {
-    expect(getShallow()).toHaveProp('target', '');
-    expect(getShallow({ newTab: true })).toHaveProp('target', '_blank');
+    expect(renderEl()).toHaveAttribute('target', '');
+    expect(renderEl({ newTab: true })).toHaveAttribute('target', '_blank');
 
-    expect(getShallow({ target: 'asdf' })).toHaveProp('target', 'asdf');
-    expect(getShallow({ target: 'asdf', newTab: true })).toHaveProp('target', 'asdf');
+    expect(renderEl({ target: 'asdf' })).toHaveAttribute('target', 'asdf');
+    expect(renderEl({ target: 'asdf', newTab: true })).toHaveAttribute('target', 'asdf');
   });
 
   it('sets the target to _blank if the external prop is true and newTab is not false, unless a different target is ' +
       'specified', function() {
-    expect(getShallow({ external: true })).toHaveProp('target', '_blank');
+    expect(renderEl({ external: true })).toHaveAttribute('target', '_blank');
+    expect(renderEl({ external: true, target: 'asdf' })).toHaveAttribute('target', 'asdf');
+
+    expect(renderEl({ external: true, newTab: true })).toHaveAttribute('target', '_blank');
+    expect(renderEl({ external: true, newTab: false })).toHaveAttribute('target', '');
 
-    expect(getShallow({ external: true, target: 'asdf' })).toHaveProp('target', 'asdf');
+    expect(renderEl({ external: true, noReferrer: true, target: 'asdf' })).toHaveAttribute('target', 'asdf');
+  });
 
-    expect(getShallow({ external: true, newTab: true })).toHaveProp('target', '_blank');
-    expect(getShallow({ external: true, newTab: false })).toHaveProp('target', '');
+  it('does not add an icon after the children if external is false', function() {
+    const { container } = quickRender({ children: <span/> });
 
-    expect(getShallow({ external: true, noReferrer: true, target: 'asdf' })).toHaveProp('target', 'asdf');
+    expect(container.querySelector('.nx-icon.fa-external-link-alt')).not.toBeInTheDocument();
   });
 
   it('adds an icon after the children if external is true', function() {
-    expect(getShallow({ children: <span/> })).not.toContainMatchingElement(NxFontAwesomeIcon);
-    expect(getShallow({ children: <span/>, external: true })).toContainMatchingElement(NxFontAwesomeIcon);
-  });
+    const { container } = quickRender({ children: 'foo', external: true });
 
-  it('wraps non-element children in an element sibling to the icon', function() {
-    expect(getShallow({ children: 'foo', external: true })).toMatchElement(
-      <a>
```

Comment:
This may still be useful to test, to check that a <span> automatically renders when text is provided (and that when the text is wrapped in something, it doesn't create a span) . I'm sure there's a cleaner way, but could be something like:    

```
const el = renderEl({children: 'foo'});
    expect(el?.children[0].tagName).toBe('SPAN');
```

Reply:
> This may still be useful to test, to check that a automatically renders when text is provided (and that when the text is wrapped in something, it doesn't create a span) . I'm sure there's a cleaner way, but could be something like:
> 
> ```
> const el = renderEl({children: 'foo'});
>     expect(el?.children[0].tagName).toBe('SPAN');
> ```

Added it here: 69ec81407e68f1a5c3c53a15bd104a400005753d